src/simulation/SimulationManager.java

```diff
@@ -405,6 +405,10 @@ public boolean performUntil(int limit) {
         return false;
     }
 
+    public void resetPathToInit() {
+        statePath.decrementPosition(statePath.getPosition(), traceMode);
```

Comment:
This is fine for traces.

However, in the case of models, this will prevent the user from obtaining alternate states. You will need to perform a "0-step" and some additional bookkeeping (see line 148 until end of function in this file) to ensure we have the A4Solution and valid internal state.